server/core/src/https/mod.rs

```diff
@@ -269,13 +270,20 @@ pub async fn create_https_server(
         ServerRole::WriteReplica | ServerRole::ReadOnlyReplica => {
             // Create a spa router that captures everything at ui without key extraction.
             if cfg!(feature = "ui_htmx") {
+                let static_compressed_router = Router::new()
+                    .layer(middleware::compression::new())
+                    .layer(from_fn(middleware::caching::cache_me_short))
+                    .route("/ui/images/oauth2/:rs_name", get(oauth2::oauth2_image_get))
+                    .route("/ui/images/domain", get(v1_domain::image_get));
+
                 Router::new()
                     .route("/", get(|| async { Redirect::to("/ui") }))
                     .nest("/ui", views::view_router())
-                    .layer(middleware::compression::new())
-                    .route("/ui/images/oauth2/:rs_name", get(oauth2::oauth2_image_get))
+                    .merge(static_compressed_router)
             } else {
                 Router::new()
+                    .route("/ui/images/oauth2/:rs_name", get(oauth2::oauth2_image_get))
+                    .layer(middleware::compression::new())
```

Comment:
The ordering of the layering here is off - the "else" arm should have the compression layer above the images endpoint - because we don't want to be wasting cycles compressing images

Reply:
Actually, the order doesn't matter. The router applies the layer as a whole to any route on the router.

Reply:
That's not what the [docs say](https://docs.rs/axum/latest/axum/middleware/index.html#applying-middleware)

> When you add middleware with Router::layer (or similar) all previously added routes will be wrapped in the middleware. Generally speaking, **this results in middleware being executed from bottom to top.**

Reply:
Argh it's so confusing. The actualy detail is here https://docs.rs/axum/latest/axum/struct.Router.html#method.layer which makes you right, but for a different reason. #2753